### ISSUE TITLE
Error access microphone & camera on browser

### ISSUE THREAD

**Issue by alexandre1985** [nextcloud/spreed#859]
## Steps to reproduce
1. Enter in Nextcloud Talk page

### Expected behaviour
Last version my camera was working on my Manjaro Firefox

### Actual behaviour
Now it doesn't work now

## Browser
**Microphone available:** yes but not working

**Camera available:** yes but not working

**Operating system:** Manjaro Linux 64-bit

**Browser name:** Firefox

**Browser version:** 59.0.2 64bit

### Browser log

<details>
```
Insert your browser log here, this could for example include:
a) The javascript console log (it is in Portuguese)
Content Security Policy: As definições da página bloquearam o carregamento de um recurso em self (“script-src”). Source: ( function () { try {

(function injectP....
spreed:1
Content Security Policy: As definições da página bloquearam o carregamento de um recurso em self (“script-src”). Source: ;(function(e){let t={};if(e.hasOwnProper....
spreed:1
Content Security Policy: As definições da página bloquearam o carregamento de um recurso em self (“script-src”). Source: ( function () {                try {
var....
spreed:1
Content Security Policy: As definições da página bloquearam o carregamento de um recurso em self (“script-src”). Source: (function (NAVIGATOR, OBJECT) {

    OBJ....
spreed:1
Content Security Policy: As definições da página bloquearam o carregamento de um recurso em self (“script-src”). Source: (function (ERROR) {

    const V8_STACK_....
spreed:1
JQMIGRATE: Migrate is installed, version 1.4.0
core.js:7:542
Content Security Policy: As definições da página bloquearam o carregamento de um recurso em self (“script-src”). Source: onclick attribute on DIV element.
spreed
Content Security Policy: As definições da página bloquearam o carregamento de um recurso em self (“script-src”). Source: ;(function(e){setTimeout(()=>{const n=do....
spreed:1
Content Security Policy: As definições da página bloquearam o carregamento de um recurso em self (“script-src”). Source: ;(function(e){let n=null,t=0;const o={no....
spreed:1
Creating one-to-one video call teste
connection.js:40:4
Joined 
Object { ocs: {…} }
signaling.js:169:5
addRoomMessage
roomlistview.js:221:4
CONNECTED
simplewebrtc.js:18020:4
startLocalVideo
simplewebrtc.js:18278:3
Access to microphone & camera failed 
MediaStreamError { name: "InternalError", message: "Starting video failed", constraint: "", stack: "" }
webrtc.js:644:4
Error while accessing microphone & camera:  Starting video failed
webrtc.js:660:5
TypeError: $element.position(...) is undefined[Saber mais]
chatview.js:406:11
video can not be enabled - there was no stream available before
app.js:196:6
Erro de mapa de fontes: request failed with status 404
URL do recurso: https://nextcloud.minho.win/core/vendor/core.js?v=f09c37a0-0
URL do mapa de fontes: purify.min.js.map[Saber mais]

b) The network log
c) ...
```

</details>

## Spreed app

**Spreed app version:** (see apps admin page: `/index.php/settings/apps`)
3.2.0

**Custom TURN server configured:** yes/no (see additional admin settings: `/index.php/settings/admin/additional`)

**Custom STUN server configured:** yes/no (see additional admin settings: `/index.php/settings/admin/additional`)

## Server configuration

**Operating system**: ArchLinux aarch64

**Web server:** Apache

**Database:** Maria

**PHP version:**
7.2.5
**Nextcloud Version:** (see admin page)
13.0.2
**List of activated apps:**

<details>
```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your server installation folder
```
Enabled:
  - activity: 2.6.1
  - bookmarks: 0.11.0
  - calendar: 1.6.1
  - comments: 1.3.0
  - contacts: 2.1.3
  - dav: 1.4.6
  - federatedfilesharing: 1.3.1
  - federation: 1.3.0
  - files: 1.8.0
  - files_pdfviewer: 1.2.1
  - files_sharing: 1.5.0
  - files_texteditor: 2.5.1
  - files_trashbin: 1.3.0
  - files_versions: 1.6.0
  - files_videoplayer: 1.2.0
  - firstrunwizard: 2.2.1
  - gallery: 18.0.0
  - groupfolders: 1.2.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.1.0
  - nextcloud_announcements: 1.2.0
  - notes: 2.3.2
  - notifications: 2.1.2
  - oauth2: 1.1.0
  - password_policy: 1.3.0
  - provisioning_api: 1.3.0
  - quota_warning: 1.2.0
  - serverinfo: 1.3.0
  - sharebymail: 1.3.0
  - spreed: 3.2.0
  - survey_client: 1.1.0
  - systemtags: 1.3.0
  - tasks: 0.9.6
  - theming: 1.4.1
  - twofactor_backupcodes: 1.2.3
  - updatenotification: 1.3.0
  - workflowengine: 1.3.0
Disabled:
  - admin_audit
  - encryption
  - files_external
  - spreed_old
  - user_external
  - user_ldap
</details>

**Nextcloud configuration:**

<details>
```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or

Insert your config.php content here
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "apps_paths": [
            {
                "path": "\/usr\/share\/webapps\/nextcloud\/apps2",
                "url": "\/apps2",
                "writable": true
            },
            {
                "path": "\/usr\/share\/webapps\/nextcloud\/apps",
                "url": "\/apps",
                "writable": false
            }
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "box",
            "nextcloud.minho.win"
        ],
        "overwrite.cli.url": "https:\/\/nextcloud.minho.win",
        "htaccess.RewriteBase": "\/",
        "dbtype": "mysql",
        "version": "13.0.2.1",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "maintenance": false,
        "memcache.local": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "dbindex": 0,
            "timeout": 1.5
        },
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "theme": "",
        "loglevel": 0
    }
}

</details>

### Server log (data/nextcloud.log)
<details>
```
{"reqId":"o8Euwavere9l54lN9nZV","level":3,"time":"2018-04-09T14:48:11+00:00","remoteAddr":"192.168.1.85","user":"--","app":"PHP","method":"POST","url":"\/nextcloud\/index.php","message":"is_dir(): open_basedir restriction in effect. File(\/mnt\/disco\/nextcloud\/) is not within the allowed path(s): (\/srv\/http\/:\/dev\/urandom:\/tmp\/:\/usr\/share\/pear\/:\/usr\/share\/webapps\/nextcloud\/:\/etc\/webapps\/nextcloud) at \/usr\/share\/webapps\/nextcloud\/lib\/private\/Setup.php#299","userAgent":"Mozilla\/5.0 (X11; Linux x86_64; rv:59.0) Gecko\/20100101 Firefox\/59.0","version":""}
{"reqId":"o8Euwavere9l54lN9nZV","level":3,"time":"2018-04-09T14:48:11+00:00","remoteAddr":"192.168.1.85","user":"--","app":"PHP","method":"POST","url":"\/nextcloud\/index.php","message":"mkdir(): open_basedir restriction in effect. File(\/mnt\/disco\/nextcloud\/) is not within the allowed path(s): (\/srv\/http\/:\/dev\/urandom:\/tmp\/:\/usr\/share\/pear\/:\/usr\/share\/webapps\/nextcloud\/:\/etc\/webapps\/nextcloud) at \/usr\/share\/webapps\/nextcloud\/lib\/private\/Setup.php#299","userAgent":"Mozilla\/5.0 (X11; Linux x86_64; rv:59.0) Gecko\/20100101 Firefox\/59.0","version":""}
{"reqId":"CnxtwY59Uamk7g5S0vLs","level":3,"time":"2018-04-09T14:49:40+00:00","remoteAddr":"192.168.1.85","user":"--","app":"PHP","method":"POST","url":"\/nextcloud\/index.php","message":"is_dir(): open_basedir restriction in effect. File(\/mnt\/disco\/nextcloud\/) is not within the allowed path(s): (\/srv\/http\/:\/dev\/urandom:\/tmp\/:\/usr\/share\/pear\/:\/usr\/share\/webapps\/nextcloud\/:\/etc\/webapps\/nextcloud) at \/usr\/share\/webapps\/nextcloud\/lib\/private\/Setup.php#299","userAgent":"Mozilla\/5.0 (X11; Linux x86_64; rv:59.0) Gecko\/20100101 Firefox\/59.0","version":""}
{"reqId":"CnxtwY59Uamk7g5S0vLs","level":3,"time":"2018-04-09T14:49:40+00:00","remoteAddr":"192.168.1.85","user":"--","app":"PHP","method":"POST","url":"\/nextcloud\/index.php","message":"mkdir(): open_basedir restriction in effect. File(\/mnt\/disco\/nextcloud\/) is not within the allowed path(s): (\/srv\/http\/:\/dev\/urandom:\/tmp\/:\/usr\/share\/pear\/:\/usr\/share\/webapps\/nextcloud\/:\/etc\/webapps\/nextcloud) at \/usr\/share\/webapps\/nextcloud\/lib\/private\/Setup.php#299","userAgent":"Mozilla\/5.0 (X11; Linux x86_64; rv:59.0) Gecko\/20100101 Firefox\/59.0","version":""}
{"reqId":"VJgyh39Foh63x0vIunsR","level":3,"time":"2018-04-09T14:52:15+00:00","remoteAddr":"192.168.1.85","user":"--","app":"PHP","method":"POST","url":"\/nextcloud\/index.php","message":"is_dir(): open_basedir restriction in effect. File(\/mnt\/disco\/nextcloud\/) is not within the allowed path(s): (\/srv\/http\/:\/dev\/urandom:\/tmp\/:\/usr\/share\/pear\/:\/usr\/share\/webapps\/nextcloud\/:\/etc\/webapps\/nextcloud) at \/usr\/share\/webapps\/nextcloud\/lib\/private\/Setup.php#299","userAgent":"Mozilla\/5.0 (X11; Linux x86_64; rv:59.0) Gecko\/20100101 Firefox\/59.0","version":""}
{"reqId":"VJgyh39Foh63x0vIunsR","level":3,"time":"2018-04-09T14:52:15+00:00","remoteAddr":"192.168.1.85","user":"--","app":"PHP","method":"POST","url":"\/nextcloud\/index.php","message":"mkdir(): open_basedir restriction in effect. File(\/mnt\/disco\/nextcloud\/) is not within the allowed path(s): (\/srv\/http\/:\/dev\/urandom:\/tmp\/:\/usr\/share\/pear\/:\/usr\/share\/webapps\/nextcloud\/:\/etc\/webapps\/nextcloud) at \/usr\/share\/webapps\/nextcloud\/lib\/private\/Setup.php#299","userAgent":"Mozilla\/5.0 (X11; Linux x86_64; rv:59.0) Gecko\/20100101 Firefox\/59.0","version":""}

```
</details>

**Comment by nickvergessen** [comment]
Should be a duplicate of #850 

You can try to apply https://github.com/nextcloud/spreed/pull/854/files manually for now and see if that helps.

**Comment by alexandre1985** [comment]
My desktop camera is not working but on my laptop everything seems ok.
Will try to apply the changes above and see what happens.

**Comment by alexandre1985** [comment]
Yes, this fixed my problem. Thank you.
One thing to note, not a major issue, but the button to join the call, after joining the call doesn't turn to "leave the call". Just so that you guys know. Not major issue.
Cheers

**Comment by nickvergessen** [comment]
Thats fixed with #858
